app/javascript/components/student/ExerciseStatusChart.tsx

```diff
@@ -7,17 +7,18 @@ import { useIsMounted } from 'use-is-mounted'
 import { FetchingBoundary } from '../FetchingBoundary'
 
 export const ExerciseStatusChart = ({
-  exerciseStatuses,
+  exercisesData,
   links,
 }: {
-  exerciseStatuses: { [slug: string]: string }
+  exercisesData: { [slug: string]: [string, string] }
```

Comment:
You could consider defining a named tuple type: [see example](https://www.typescriptlang.org/play?ts=4.0.2#code/PTAEBUFcAcBsFMDOoCGAneq1pQT2QO4AW8GoALiaAPZoAmpoAlskwLbS3koB25F1ClXK5omRPnLw2AGgCwAKBChc1SKADGvUAnQ9QbWphQAjNf0rTm++AA8UHBAC5yMBIkWLlASX3hR8ADKGmhM0PwALAB0AAwyQpgiYjQAZqgQbvAA5MgA5ihMPPB0CagmTLBMIgKguUwAbvBeYCigPA6JgpagdEwpKaTwfKDQ6OTI1Gnd6Di4UZ5KYABitKB2DnDw8arqkIjFNQShUukAMij8p9Q8uTrUWuRM16D1TK2trptOC0mYAPKwOhXB5PfQAXlAAG0eJA2CZSPEYXDSABdBYaa6Ifiwe4XUGIJygAFA3GPa6QlGgCGQxSgOlQiIxKIAdgAjBEIvEALTM6IxGIANhR8gU9KhAFYAMxRAURcUADm5rJZ8vFaIU6uaoAA0jxqARCrdiEwNERmMhzo9XAxUDwSlcblVIDaWKg4UxcpA1Ih4rwSohBDstQQ1ICDEYdEwANbwWC4UBEFCNTQoWAIEpVM4XB23T4IeYKLUAdSqZr52zUKf0Ryq8G+hYUv1AADl4ARgXjntTYBdCUj4Wh4jibn3YQPNQoMTwsW02x2ydPCa326TQRSqVDafTIeKAExRSXMmK7+LReUATj3wq3dJ30slEXPitAXIFUV3ApikvVE+U4Co7RsEgbT6qAiBEKBMDWKUxRVKsxBDColYQY0aA0KhWrdDEtolKyqAWFQQwlJMpRFLY2KFE0k6YvwKRMGgWKWhuRQriCmKQjEKIceqU4znRDHkDmzFzqu7GcZCrK-mARZEBUiSJvwbAekQ-D7FYHwoCUkB2qQCGwEpNzxN0bAFPouTUKmoBaq65CCEMiCQGQlAXNYKS0CZC7mjwWTYtQM4IdWtBRoawalqUTYSFi0hRKAKxoesjhbKAAVrOROAaI8NxaqMOBAVIDGgCkaDUGw6QpNpGWgqAeyGqUAAK6AdPlVmLNVjyVNUvz1oo5U8JVzwaEM+UALIoNAAAUsAjm0Y4IjovYzciaAAJSgAA3gAvgsPj6OWQiujG8DQMgU23H683kIoTYAMJDaQo3QA1uXIBCT1NaQiAADy-CRg18PdY0AHzbWAvigNK55GbJhChiUOLUFGkYxldAREoCt3-WgD1vWwL1Qv2c0E2gUmgKNwU3GBJWYCRxkRhiiW2BQqMmTwYSQD2nk4ogEham5aHdDl71oYUfMeaCURAA)

This will make the code below where the value is extracted from the tuple a bit nicer.

Reply:
Thanks! 🙂  I decided not to do that here as I don't want the extra loads of chars being passed over in the JSON (it would add an extra 2k to the size of the data for the C# track). I did do this for the concept map version though.